sensor/kubernetes/listener/resources/selector_test.go

```diff
@@ -0,0 +1,215 @@
+package resources
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+var (
+	labelsEmpty          = map[string]string{}
+	labelsOneElement     = map[string]string{"FirstEle": "FirstEleValue"}
+	labelsThreeElements  = map[string]string{"FirstEle": "FirstEleValue", "2nd": "2ndValue", "3rd": "3rdValue"}
+	labelsThreeElements2 = map[string]string{"4th": "Val4", "5th": "Val5", "6th": "Val6"}
+	labelsFiveElements   = map[string]string{"1": "2", "2": "3", "3": "4", "4": "5", "5": "6"}
+)
+
+var _ suite.SetupTestSuite = (*SelectorWrapperTestSuite)(nil)
+
+func TestSelectorWrapper(t *testing.T) {
+	suite.Run(t, new(SelectorWrapperTestSuite))
+}
+
+func (s *SelectorWrapperTestSuite) SetupTest() {
+	s.hasMatchesBeenCalled = false
+}
+
+type SelectorWrapperTestSuite struct {
+	suite.Suite
+	hasMatchesBeenCalled bool
+}
+
+type mockSelector struct {
+	internalSelector labels.Selector
+	testSuite        *SelectorWrapperTestSuite
+}
+
+func (m mockSelector) Empty() bool {
+	return false
+}
+
+func (m mockSelector) String() string {
+	return ""
+}
+
+func (m mockSelector) Add(r ...labels.Requirement) labels.Selector {
+	return nil
+}
+
+func (m mockSelector) Requirements() (requirements labels.Requirements, selectable bool) {
+	return nil, false
+}
+
+func (m mockSelector) DeepCopySelector() labels.Selector {
+	return nil
+}
+
+func (m mockSelector) RequiresExactMatch(label string) (value string, found bool) {
+	return "", false
+}
+
+func (m mockSelector) Matches(labels labels.Labels) bool {
+	m.testSuite.hasMatchesBeenCalled = true
+	return m.internalSelector.Matches(labels)
+}
+
+func (s *SelectorWrapperTestSuite) injectMockSelector(sw *selectorWrap) {
+	sw.selector = mockSelector{sw.selector, s}
+}
+
+func (s *SelectorWrapperTestSuite) TestLabelMatching() {
+	tests := map[string]struct {
+		givenSelectorLabels                         map[string]string
+		matchEmptySelector                          selectorWrapOption
+		givenMatchingLabels                         map[string]string
+		expectedMatch                               bool
+		expectedInternalSelectorMatchesFunctionCall bool
```

Comment:
I was thinking about something like:
- `expectFullMatch`
- `expectMatchUsingNumLabelsOnly`
- ...

Reply:
To me those names are less clear than the current name, so I'll leave it as is for now. If someone else reads through this PR please comment on what you find to be clearer.

Reply:
🤔  I think that ultimately both are good but gun to my head I would choose:
```golang
tests := map[string]struct {
		givenSelectorLabels          map[string]string
		matchEmptySelector           selectorWrapOption
		givenMatchingLabels          map[string]string
		expectedMatch                bool
		expectedMatchUsingNumLabels  bool
}
```